db/statistics.go

```diff
@@ -26,266 +29,435 @@ func WriteValidatorStatisticsForDay(day uint64) error {
 	}
+	logger.Infof("validating completed, took %v", time.Since(start))
 
+	tx, err := WriterDb.Beginx()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	start = time.Now()
+
+	logger.Infof("exporting total income stats")
+	_, err = tx.Exec(`
+	INSERT INTO validator_stats (validatorindex, day, cl_rewards_gwei_total, cl_proposer_rewards_gwei_total, el_rewards_wei_total, mev_rewards_wei_total) (
+		SELECT 
+			vs1.validatorindex, 
+			vs1.day, 
+			COALESCE(vs1.cl_rewards_gwei, 0) + COALESCE(vs2.cl_rewards_gwei_total, 0) AS cl_rewards_gwei_total_new, 
+			COALESCE(vs1.cl_proposer_rewards_gwei, 0) + COALESCE(vs2.cl_proposer_rewards_gwei_total, 0) AS cl_proposer_rewards_gwei_total_new, 
+			COALESCE(vs1.el_rewards_wei, 0) + COALESCE(vs2.el_rewards_wei_total, 0) AS el_rewards_wei_total_new, 
+			COALESCE(vs1.mev_rewards_wei, 0) + COALESCE(vs2.mev_rewards_wei_total, 0) AS mev_rewards_wei_total_new 
+		FROM validator_stats vs1 LEFT JOIN validator_stats vs2 ON vs2.day = vs1.day - 1 AND vs2.validatorindex = vs1.validatorindex WHERE vs1.day = $1
+	) ON CONFLICT (validatorindex, day) DO UPDATE SET 
+		cl_rewards_gwei_total = excluded.cl_rewards_gwei_total,
+		cl_proposer_rewards_gwei_total = excluded.cl_proposer_rewards_gwei_total,
+		el_rewards_wei_total = excluded.el_rewards_wei_total,
+		mev_rewards_wei_total = excluded.mev_rewards_wei_total;
+	`, day)
+	if err != nil {
+		return err
+	}
 	logger.Infof("export completed, took %v", time.Since(start))
 
 	start = time.Now()
 
-	logger.Infof("exporting deposits and deposits_amount statistics")
-	depositsQry := `
-		insert into validator_stats (validatorindex, day, deposits, deposits_amount) 
-		(
-			select validators.validatorindex, $3, count(*), sum(amount)
-			from blocks_deposits
-			inner join validators on blocks_deposits.publickey = validators.pubkey
-			inner join blocks on blocks_deposits.block_root = blocks.blockroot
-			where blocks.epoch >= $1 and blocks.epoch <= $2 and blocks.status = '1' and blocks_deposits.valid_signature
-			group by validators.validatorindex
+	logger.Infof("populate validator_performance table")
+	_, err = tx.Exec(`insert into validator_performance (
+		validatorindex,
+		balance,
+		performance1d,
+		performance7d,
+		performance31d,
+		performance365d,
+
+		rank7d,
+
+		cl_performance_1d,
+		cl_performance_7d,
+		cl_performance_31d,
+		cl_performance_365d,
+		cl_performance_total,
+		cl_proposer_performance_total,
+
+		el_performance_1d,
+		el_performance_7d,
+		el_performance_31d,
+		el_performance_365d,
+		el_performance_total,
+
+		mev_performance_1d,
+		mev_performance_7d,
+		mev_performance_31d,
+		mev_performance_365d,
+		mev_performance_total
+		) (
+			select 
+			vs_now.validatorindex, 
+				COALESCE(vs_now.end_balance, 0) as balance, 
```

Comment:
This query is dependent on the balance being exported. This is not reflected here though:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/401df27e99a88a6138ed1b8b8ee3911e93a67fba/db/statistics.go#L184-L201